### DISCUSSION TITLE
Per-project plugins

### DISCUSSION THREAD

**Discussion by MrQubo** [Vector35/binaryninja-api#4763] [Ideas] [answered]
I have many plugins installed but some of them (e.g. golang plugins) I only use in some projects. All those plugins end up clobbering UI. It would be nice to be able to disable some of the plugins in user settings and enable them in project setting, when needed.

**Comment by plafosse** [comment, marked as answer]
I converted this to an issue. https://github.com/Vector35/binaryninja-api/issues/4781

**Comment by psifertex** [comment]
Here's a work-around that can be very effective: 

You can use the BN_USER_DIRECTORY environment variable to create different profiles with different plugins installed or even settings. I don't think our current implementation will support different plugins per project within the same process, but the environment variable offers a really useful way to get the same effect. 

https://docs.binary.ninja/guide/plugins.html#troubleshooting

**Reply by MrQubo** [reply]
Thanks, that's a good idea. I already have per-project nix configs and I only need to add one line in there.
